plugins/inputs/sqlserver/sqlserverqueries.go

```diff
@@ -1158,48 +1158,61 @@ IF SERVERPROPERTY('EngineEdition') NOT IN (2,3,4) BEGIN /*NOT IN Standard,Enterp
 	RETURN
 END
 
-IF SERVERPROPERTY('IsHadrEnabled') = 1 BEGIN
-	SELECT
-		'sqlserver_hadr_replica_states' AS [measurement],
-		REPLACE(@@SERVERNAME,'\',':') AS [sql_instance],
-		convert(nvarchar(36), hars.replica_id) as replica_id,
-		ar.replica_server_name,
-		convert(nvarchar(36), hars.group_id) as group_id,
-		ag.name AS group_name,
-		ag.basic_features,
-		ag.is_distributed,
-		hags.synchronization_health_desc AS ag_synchronization_health_desc,
-		ar.replica_metadata_id,
-		ar.availability_mode,
-		ar.availability_mode_desc,
-		ar.failover_mode,
-		ar.failover_mode_desc,
-		ar.session_timeout,
-		ar.primary_role_allow_connections,
-		ar.primary_role_allow_connections_desc,
-		ar.secondary_role_allow_connections,
-		ar.secondary_role_allow_connections_desc,
-		ar.seeding_mode,
-		ar.seeding_mode_desc,
-		hars.is_local,
-		hars.role,
-		hars.role_desc,
-		hars.operational_state,
-		hars.operational_state_desc,
-		hars.connected_state,
-		hars.connected_state_desc,
-		hars.recovery_health,
-		hars.recovery_health_desc,
-		hars.synchronization_health AS replica_synchronization_health,
-		hars.synchronization_health_desc AS replica_synchronization_health_desc,
-		hars.last_connect_error_number,
-		hars.last_connect_error_description,
-		hars.last_connect_error_timestamp
-	from sys.dm_hadr_availability_replica_states AS hars
-	inner join sys.availability_replicas AS ar on hars.replica_id = ar.replica_id
-	inner join sys.availability_groups AS ag on ar.group_id = ag.group_id
-	inner join sys.dm_hadr_availability_group_states AS hags ON hags.group_id = ag.group_id
+DECLARE
+	 @SqlStatement AS nvarchar(max)
+	,@MajorMinorVersion AS int = CAST(PARSENAME(CAST(SERVERPROPERTY('ProductVersion') AS nvarchar),4) AS int)*100 + CAST(PARSENAME(CAST(SERVERPROPERTY('ProductVersion') AS nvarchar),3) AS int)
+	,@Columns AS nvarchar(MAX) = ''
+
+IF @MajorMinorVersion >= 1300 BEGIN
```

Comment:
Thank you for the fix! Can you also add/update the unit tests for AG queries in sqlserver_test.go?

Reply:
Added couple of test cases for AG queries.

Reply:
Thanks! Please mark it as resolved!